x-pack/plugins/task_manager/server/task_claimers/strategy_mget.ts

```diff
@@ -0,0 +1,366 @@
+
+  const finalResult = {
+    stats: {
+      tasksUpdated: finalResults.length,
+      tasksConflicted: conflicts,
+      tasksClaimed: finalResults.length,
+    },
+    docs: finalResults,
+    timing: stopTaskTimer(),
+  };
+
+  for (const doc of finalResults) {
+    events$.next(asTaskClaimEvent(doc.id, asOk(doc), finalResult.timing));
+  }
+
+  return finalResult;
+}
+
+interface SearchAvailableTasksResponse {
+  docs: ConcreteTaskInstance[];
+  versionMap: Map<string, ConcreteTaskInstanceVersion>;
+}
+
+async function searchAvailableTasks({
+  definitions,
+  taskTypes,
+  removedTypes,
+  excludedTypes,
+  taskStore,
+  claimOwnershipUntil,
+  size,
+  taskMaxAttempts,
+}: OwnershipClaimingOpts): Promise<SearchAvailableTasksResponse> {
+  const searchedTypes = Array.from(taskTypes)
+    .concat(Array.from(removedTypes))
+    .filter((type) => !excludedTypes.has(type));
+  const queryForScheduledTasks = mustBeAllOf(
+    // Task must be enabled
+    EnabledTask,
+    // a task type that's not excluded (may be removed or not)
+    OneOfTaskTypes('task.taskType', searchedTypes),
+    // Either a task with idle status and runAt <= now or
+    // status running or claiming with a retryAt <= now.
+    shouldBeOneOf(IdleTaskWithExpiredRunAt, RunningOrClaimingTaskWithExpiredRetryAt),
+    // must have a status that isn't 'unrecognized'
+    RecognizedTask
+  );
+
+  const sort: NonNullable<SearchOpts['sort']> = getClaimSort(definitions);
+  const query = matchesClauses(queryForScheduledTasks, filterDownBy(InactiveTasks));
+
+  return await taskStore.fetch({
+    query,
+    sort,
+    size,
+    seq_no_primary_term: true,
+  });
+}
+
+function getClaimSort(definitions: TaskTypeDictionary): estypes.SortCombinations[] {
+  // Sort by descending priority, then by ascending runAt/retryAt time
+  if (definitions.size() === 0) {
+    return [SortByRunAtAndRetryAt];
+  }
+
+  return [
+    {
+      _script: {
+        type: 'number',
+        order: 'desc',
+        script: {
+          lang: 'painless',
+          // Use priority if explicitly specified in task definition, otherwise default to 50 (Normal)
+          // TODO: we could do this locally as well, but they may starve
+          source: `
+            String taskType = doc['task.taskType'].value;
+            if (params.priority_map.containsKey(taskType)) {
+              return params.priority_map[taskType];
+            } else {
+              return ${TaskPriority.Normal};
+            }
+          `,
+          params: {
+            priority_map: definitions
+              .getAllDefinitions()
+              .reduce<Record<string, TaskPriority>>((acc, taskDefinition) => {
+                if (taskDefinition.priority) {
+                  acc[taskDefinition.type] = taskDefinition.priority;
+                }
+                return acc;
+              }, {}),
+          },
+        },
+      },
+    },
+    SortByRunAtAndRetryAt,
+  ];
+}
+
+function emptyClaimOwnershipResult() {
```

Comment:
this looks like the same sort as in strategy_default (with the addition of the size check). should we move this out into into a shared location and reuse?

Reply:
moving this into the `index.ts` file in this module's directory - I assume the reference to the `sort` was  some copy/pasta and you intended `emptyClaimOwnershipResult()`.